src/transport/SecureSessionMgr.h

```diff
@@ -92,13 +97,18 @@ class EncryptedPacketBufferHandle final : private System::PacketBufferHandle
     CHIP_ERROR InsertPacketHeader(const PacketHeader & aPacketHeader) { return aPacketHeader.EncodeBeforeData(*this); }
 #endif // CHIP_ENABLE_TEST_ENCRYPTED_BUFFER_API
 
-private:
-    // Allow SecureSessionMgr to assign or construct us from a PacketBufferHandle
-    friend class SecureSessionMgr;
+    void MarkEncrypted(PacketBufferHandle && aBuffer)
+    {
+        PacketBufferHandle::operator=(std::move(aBuffer));
+        mEncryptedPacketBuffer      = true;
+    }
+
+    bool IsEncrypted() const { return mEncryptedPacketBuffer; }
 
+private:
     EncryptedPacketBufferHandle(PacketBufferHandle && aBuffer) : PacketBufferHandle(std::move(aBuffer)) {}
 
-    void operator=(PacketBufferHandle && aBuffer) { PacketBufferHandle::operator=(std::move(aBuffer)); }
+    bool mEncryptedPacketBuffer = false;
```

Comment:
The class name says "EncryptedPacketBufferHandle". Having a bool checking again if encrypted or not seems odd.

Can we update this in a different PR together with some renaming that makes the usage more logical?